src/Spectre.libPlatform/Math.h

```diff
@@ -0,0 +1,170 @@
+}
+
+template< class DataType >
+constexpr std::vector<DataType> times(gsl::span<const DataType> first, gsl::span<const DataType> second)
+{
+    return Functional::transform(first, second, [](DataType left, DataType right) { return left * right; });
+}
+
+template< class DataType >
+constexpr std::vector<DataType> divideBy(gsl::span<const DataType> first, gsl::span<const DataType> second)
+{
+    return Functional::transform(first, second, [](DataType left, DataType right) { return left / right; });
+}
+
+template< class DataType >
+constexpr std::vector<DataType> modulo(gsl::span<const DataType> first, gsl::span<const DataType> second)
+{
+    return Functional::transform(first, second, [](DataType left, DataType right) { return left % right; });
+}
+
+template< class DataType >
+constexpr std::vector<DataType> bitwiseAnd(gsl::span<const DataType> first, gsl::span<const DataType> second)
+{
+    return Functional::transform(first, second, [](DataType left, DataType right) { return left & right; });
+}
+
+template< class DataType >
+constexpr std::vector<DataType> logicalAnd(gsl::span<const DataType> first, gsl::span<const DataType> second)
+{
+    return Functional::transform(first, second, [](DataType left, DataType right) { return left && right; });
+}
+
+template< class DataType >
+constexpr std::vector<DataType> bitwiseOr(gsl::span<const DataType> first, gsl::span<const DataType> second)
+{
+    return Functional::transform(first, second, [](DataType left, DataType right) { return left | right; });
+}
+
+template< class DataType >
+constexpr std::vector<DataType> logicalOr(gsl::span<const DataType> first, gsl::span<const DataType> second)
+{
+    return Functional::transform(first, second, [](DataType left, DataType right) { return left || right; });
+}
+
+template<class DataType>
+constexpr std::vector<DataType> max(gsl::span<const DataType> first, gsl::span<const DataType> second)
+{
+    return Functional::transform(first, second, [](DataType left, DataType right) { return std::max(left, right); });
+}
+
+template<class DataType>
+constexpr std::vector<DataType> min(gsl::span<const DataType> first, gsl::span<const DataType> second)
+{
+    return Functional::transform(first, second, [](DataType left, DataType right) { return std::min(left, right); });
+}
+
+// @gmrukwa: vector vs scalar
+
+template< class DataType >
+constexpr std::vector<DataType> plus(gsl::span<const DataType> first, DataType second)
+{
+    return Functional::transform(first, [second](DataType left) { return left + second; });
+}
+
+template< class DataType >
+constexpr std::vector<DataType> minus(gsl::span<const DataType> first, DataType second)
+{
+    return Functional::transform(first, [second](DataType left) { return left - second; });
+}
+
+template< class DataType >
+constexpr std::vector<DataType> times(gsl::span<const DataType> first, DataType second)
```

Comment:
"Times" may do its job, but if one seeks for multiplication operator "multiply" or "multiplyBy" may be better in my opinion.